### ISSUE TITLE
Failed at pre-evaluation when training with MotifPredictor for SGCLS & SGDET

### ISSUE THREAD

**Issue by juliuswang0728** [KaihuaTang/Scene-Graph-Benchmark.pytorch#8]
Hi, 

This happened during pre-evaluation (`SOLVER.PRE_VAL = True`) before the training started. I adopted the [config file](https://github.com/KaihuaTang/Scene-Graph-Benchmark.pytorch/blob/master/configs/e2e_relation_X_101_32_8_FPN_1x.yaml) with `PREDICTOR: "MotifPredictor", adapting it for SGCLS or SGDET`, i.e. 

`MODEL.ROI_RELATION_HEAD.USE_GT_BOX True MODEL.ROI_RELATION_HEAD.USE_GT_OBJECT_LABEL False`, or

`MODEL.ROI_RELATION_HEAD.USE_GT_BOX False MODEL.ROI_RELATION_HEAD.USE_GT_OBJECT_LABEL False`,

the following assertion was triggered

https://github.com/KaihuaTang/Scene-Graph-Benchmark.pytorch/blob/0482daef90c98f810774aab042005add6859f529/maskrcnn_benchmark/modeling/roi_heads/relation_head/model_motifs.py#L178

It seems the evaluation can proceed if I comment out the line, I'll let the training run and see where it leads me to. So, still not sure if that's an issue/bug/etc?

best,
Julius

**Comment by KaihuaTang** [comment]
This part of the code is copied from neural-motifs. Original code just wants to ensure the test/eval mode only runs on batch_size == 1, otherwise, it may trigger some errors in neural-motifs project. 

I didn't comment out this line only because I want to make sure its code running correctly, so I set all TEST.IMS_PER_BATCH == 1 (batch_size during test/eval).

**Comment by juliuswang0728** [comment]
I've confirmed that It requires `TEST.IMS_PER_BATCH == NUM_GPUS`. 
Thanks for your speedy response, I'll close this.
